2.4/test/test_s2i.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python3
+import logging
+import os
+import time
+
+from conu import DockerBackend, S2IDockerImage, Probe, DockerRunBuilder
+
+import pytest
+
+
+image_name = os.environ.get("IMAGE_NAME", "ruby")
+test_dir = os.path.abspath(os.path.dirname(__file__))
+puma_app_path = os.path.join(test_dir, "puma-test-app")
+rack_app_path = os.path.join(test_dir, "rack-test-app")
+app_paths = [
+    puma_app_path,
+    rack_app_path
+]
+
+
+backend = DockerBackend(logging_level=logging.DEBUG)
+
+
+@pytest.fixture(scope="module", params=app_paths)
+def app(request):
+    i = S2IDockerImage(image_name)
+    app_name = os.path.basename(request.param)
+    app = i.extend(request.param, app_name)
+    yield app
+    pass
+    app.rmi()
+
+
+class TestSuite:
+    def test_s2i_apps(self, app):
+        c = app.run_via_binary()
+        try:
+            c.wait_for_port(8080)
+            assert c.is_port_open(8080)
+            response = c.http_request("/", port="8080")
+            assert response.ok
+            output = c.execute(["ruby", "--version"])
+            assert "ruby 2.4." in output.decode("utf-8")
+        finally:
+            c.stop()
+            c.wait()
+            # debugging
+            print(c.logs())
+            c.delete()
+
+    def test_invoking_container(self):
+        image = backend.ImageClass(image_name)
+        c = image.run_via_binary(DockerRunBuilder(command=["ruby", "--version"]))
+        try:
+            c.wait()
+            assert "ruby 2.4." in c.logs().decode("utf-8")
+        finally:
+            c.stop()
+            c.wait()
+            c.delete()
```

Comment:
This workflow of `wait`, `assert`, `stop`, `wait`, `delete` seems to be to be used pretty often. We might have a method `image.check_output(["ruby", "--version"], "ruby 2.4.")`. And the whole test case could be shortened to:
```
backend.ImageClass(image_name).check_output(command=["ruby", "--version"], matches="*ruby 2.4.*")
```
...or something along those lines..